From 76e9cf46d6e682798a2f3c9cdd6e24cff93ad661 Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Date: Thu, 20 Jun 2024 11:02:44 -0700
QSDK: wifi: mac80211: Add HW feature flag to indicate Direct
 Switching support

Add Hardware feature flag to indicate support for Direct Switching.
Drivers use this to advertize the capability.

Patch-depdendency: None

Signed-off-by: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
---
 include/net/mac80211.h | 3 +++
 net/mac80211/debugfs.c | 1 +
 2 files changed, 4 insertions(+)

diff --git a/include/net/mac80211.h b/include/net/mac80211.h
index b0456c84cfbb..34f5d5f129a8 100644
--- a/include/net/mac80211.h
+++ b/include/net/mac80211.h
@@ -2809,6 +2809,8 @@ struct ieee80211_txq {
  * @IEE80211_HW_HAS_TX_QUEUE: Hardware/drivers has tx queue, does skb queuing itself,
  *	the stack will not do tx queuing.
  *
+ * @IEE80211_HW_SUPPORT_ECM_REGISTRATION: driver supports ECM registration
+ *
  * @NUM_IEEE80211_HW_FLAGS: number of hardware flags, used for sizing arrays
  */
 enum ieee80211_hw_flags {
@@ -2871,6 +2873,7 @@ enum ieee80211_hw_flags {
 	IEEE80211_HW_SUPPORTS_TID_CLASS_OFFLOAD,
 	IEEE80211_HW_SUPPORTS_AP_PS,
 	IEEE80211_HW_HAS_TX_QUEUE,
+	IEEE80211_HW_SUPPORT_ECM_REGISTRATION,
 
 	/* keep last, obviously */
 	NUM_IEEE80211_HW_FLAGS
diff --git a/net/mac80211/debugfs.c b/net/mac80211/debugfs.c
index 06c848ed52b5..9962108a2b21 100644
--- a/net/mac80211/debugfs.c
+++ b/net/mac80211/debugfs.c
@@ -573,6 +573,7 @@ static const char *hw_flag_names[] = {
 	FLAG(SUPPORTS_TID_CLASS_OFFLOAD),
 	FLAG(SUPPORTS_AP_PS),
 	FLAG(HAS_TX_QUEUE),
+	FLAG(SUPPORT_ECM_REGISTRATION),
 #undef FLAG
 };
 
-- 
2.34.1

